mr/src/main/java/org/elasticsearch/hadoop/rest/RestClient.java

```diff
@@ -588,12 +602,15 @@ public void putMapping(String index, String mapping, byte[] bytes) {
         execute(PUT, mapping, new BytesArray(bytes));
     }
 
-    public String esVersion() {
-        Map<String, String> version = get("", "version");
-        if (version == null || !StringUtils.hasText(version.get("number"))) {
-            return "Unknown";
+    public synchronized EsVersion esVersion() {
```

Comment:
I think this should be reverted back to being retrieved from the configuration at construction time. This method is called many times in the client and the synchronized nature of it is not great for performance. There really shouldn't be a scenario where the value of the version will be different between tasks, so it's reasonable for the version information to be stored once for the entire job.